### ISSUE TITLE
Caddyfile nested imports problem

### ISSUE THREAD

**Issue by kotsius** [caddyserver/caddy#4914]
```
(nested) {
	header server pseudo
}

(problem) {
	import nested
	reverse_proxy {args.0}
}

(serve) {
	handle {
		import problem {args.0}
	}
}

www.example.com {
	import serve 10.10.10.10:8080
}
```

The above Caddyfile only validates when the “import nested” line is commented out, or when the handle block is stripped from (serve). As Francis Lavoie remarked, the import should already be expanded before handle gets to see it.

Log output is: `validate: adapting config using caddyfile: parsing caddyfile tokens for ‘handle’: Caddyfile:13 - Error during parsing: unrecognized directive: import - are you sure your Caddyfile structure (nesting and braces) is correct?`

**Comment by francislavoie** [comment]
I'm not able to replicate this anymore. Sorry we left this hanging for so long :see_no_evil: but it seems to work fine now.

```
$ caddy adapt --config Caddyfile | jq
{
  "apps": {
    "http": {
      "servers": {
        "srv0": {
          "listen": [
            ":443"
          ],
          "routes": [
            {
              "match": [
                {
                  "host": [
                    "www.example.com"
                  ]
                }
              ],
              "handle": [
                {
                  "handler": "subroute",
                  "routes": [
                    {
                      "handle": [
                        {
                          "handler": "subroute",
                          "routes": [
                            {
                              "handle": [
                                {
                                  "handler": "headers",
                                  "response": {
                                    "set": {
                                      "Server": [
                                        "pseudo"
                                      ]
                                    }
                                  }
                                },
                                {
                                  "handler": "reverse_proxy",
                                  "upstreams": [
                                    {
                                      "dial": "10.10.10.10:8080"
                                    }
                                  ]
                                }
                              ]
                            }
                          ]
                        }
                      ]
                    }
                  ]
                }
              ],
              "terminal": true
            }
          ]
        }
      }
    }
  }
}
```

**Comment by kotsius** [comment]
Having written a Caddyfile parser (in Elixir) as a workaround, the delay has not affected me at all. The issue however persists. Running 'caddy validate' on Caddy v2.6.4 with my original Caddyfile (quoted above) returns:

![sCapture-787](https://user-images.githubusercontent.com/35947133/221596149-604d5b61-6941-4c67-8d28-f4601655e81d.png)

**Comment by francislavoie** [comment]
Ah yeah, you're right, my bad. I tested using a build from `master` and not with v2.6.4.

I think it was fixed incidentally by 8bc05e598da0068358a2876c45f92f2c77455341
